Add pausing deployment rollouts

tom1299 opened the pull request:
Note that this pull request is just a draft for implementing the functionality described in the enhancement https://github.com/stakater/Reloader/issues/837

tom1299 commented on internal/pkg/handler/upgrade.go:
```diff
@@ -274,6 +279,58 @@ func PerformAction(clients kube.Clients, config util.Config, upgradeFuncs callba
+
+								time.AfterFunc(pauseDuration, func() {
+									deployment, err := clients.KubernetesClient.AppsV1().Deployments(itemNamespace).Get(context.TODO(), itemName, metav1.GetOptions{})
+									if err != nil {
```
@karl-johan-grahn Thanks for the review. The ode is really not that good. As I wrote in the issue this was just code to prove that it works. Would it be ok to add such a feature in general ? I would then of course refactor the code and add appropriate tests.

karl-johan-grahn replied:
Pull requests are very much welcome! But we will not add any temporary or work-in-progress changes, so you need to work on it until it is production-ready. Thousands of customers use the tool in production so it needs to go through thorough work before being merged.

tom1299 replied:
@karl-johan-grahn Of course. I would just need a little review / hints what might still be enhanced. Do you have any suggestions ?

karl-johan-grahn replied:
We will review it and you have to be a little patient, we only spend around 1 h each week on all open source repositories

galkindmitrii commented:
Hi@karl-johan-grahn @tom1299

Any chance on getting this reviewed and merged?

That feature will be super helpful :)

Thanks!

tom1299 commented:
@galkindmitrii @karl-johan-grahn I would be happy to continue working on this. I made some smaller refactorings but would now say it could be reviewed. Regarding documentation I am not sure where I should add it.
Any feedback is welcome 🙂

tom1299 commented:
@karl-johan-grahn I was wondering if we might still be able to get this pull request merged. I've taken some time to rebase the branch on master, refactored the code to support patching resources, moved most of the logic into a dedicated Go file, and added some additional tests.
I understand that your time is limited, and this is a fairly large PR, so I really appreciate any feedback. Thanks in advance for having a look.
